synq/video.go

```diff
@@ -162,6 +212,63 @@ func (v *Video) UploadFile(fileName string) error {
 	return nil
 }
 
+// MultipartUpload uploads a file as the video's original_file.
+// This procedure will use Amazon S3's Multipart Upload API:
+// http://docs.aws.amazon.com/AmazonS3/latest/dev/uploadobjusingmpu.html
+//
+// Example:
+//     var err error
+//     var video synq.Video
+//
+//     api := synq.New("00000000000000000000000000000000")
+//
+//     // Create a new video object.
+//     video, err = api.Create()
+//     if err != nil {
+//             log.Println(err)
+//             os.Exit(-1)
+//     }
+//
+//     // Upload the file video.mp4 in multiple parts.
+//     err = video.MultipartUpload("video.mp4")
+//     if err != nil {
+//             log.Println(err)
+//             os.Exit(-1)
+//     }
+func (v *Video) MultipartUpload(fileName string) error {
+	if err := v.GetUploaderInfo(); err != nil {
+		log.Println("failed to getUploaderInfo()")
+		return err
+	}
+
+	// NOTE(mastensg): getUploadInfo for parameters: acl, bucket, contentType, key
+	if err := v.GetUploadInfo(); err != nil {
```

Comment:
the parameters `acl`, `bucket`, `contentType`, and `key` are not returned by `v1/video/uploader`, but i think that the parameters returned from `v1/video/upload` can be used for multipart uploading

this relies on assumptions that are not guaranteed by the network api. i wish that https://github.com/SYNQfm/obaku/issues/1043 would be done, so that this workaround could be avoided

i'm sorry if this doesn't directly answer your question